util/tracecpu/profile.go

```diff
@@ -0,0 +1,437 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracecpu
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime/pprof"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/google/pprof/profile"
+	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+const (
+	labelSQL        = "sql"
+	labelSQLDigest  = "sql_digest"
+	labelPlanDigest = "plan_digest"
+)
+
+// GlobalSQLStatsProfiler is the global SQL stats profiler.
+var GlobalSQLStatsProfiler = NewSQLStatsProfiler()
+
+type sqlStatsProfiler struct {
+	taskCh     chan *profileTask
+	cacheBufCh chan *profileTask
+
+	mu struct {
+		sync.Mutex
+		ept *exportProfileTask
+	}
+	collector TopSQLCollector
```

Comment:
Maybe needs a sync.Value to aviod race condition?

Reply:
Nice catch.